### ISSUE TITLE
OneDrive for Business: Access token has expired

### ISSUE THREAD

**Issue by pstorch** [PhilippC/keepass2android#983]
I had a successfull synchronize with OneDrive for business right after I updated the app to 1.07b-pre2.

Today I tried it again and got this:

```
2019-10-07 18:26:52.882 10142-25151/keepass2android.keepass2android D/KP2A: Status Code: Unauthorized
    Microsoft.Graph.ServiceException: Code: InvalidAuthenticationToken
    Message: Access token has expired.
    
    Inner error
    
      at Microsoft.Graph.HttpProvider+<SendAsync>d__18.MoveNext () [0x001e4] in <6e1866892f3346e7ba899a6dc936762c>:0 
    --- End of stack trace from previous location where exception was thrown ---
      at Microsoft.Graph.BaseRequest+<SendRequestAsync>d__35.MoveNext () [0x001c6] in <6e1866892f3346e7ba899a6dc936762c>:0 
    --- End of stack trace from previous location where exception was thrown ---
      at Microsoft.Graph.BaseRequest+<SendStreamRequestAsync>d__33.MoveNext () [0x00088] in <6e1866892f3346e7ba899a6dc936762c>:0 
    --- End of stack trace from previous location where exception was thrown ---
      at keepass2android.Io.OneDrive2FileStorage`1[OneDrive2PrefixContainerType].OpenFileForRead (KeePassLib.Serialization.IOConnectionInfo ioc) [0x00055] in <00b10d12b62e4f58b835619475f75697>:0 
      at keepass2android.Io.OfflineSwitchableFileStorage.OpenFileForRead (KeePassLib.Serialization.IOConnectionInfo ioc) [0x00006] in <00b10d12b62e4f58b835619475f75697>:0 
      at keepass2android.Io.CachingFileStorage.GetRemoteDataAndHash (KeePassLib.Serialization.IOConnectionInfo ioc, System.String& hash) [0x00006] in <00b10d12b62e4f58b835619475f75697>:0 
      at keepass2android.SynchronizeCachedDatabase.Run () [0x0005b] in <00b10d12b62e4f58b835619475f75697>:0 

```

How can I reauthenticate the App?

**Comment by PhilippC** [comment]
can you please try to close the database (fully, i.e. not only QuickUnlock) and then reopen it. I think that should refresh the token. But of course that's just a workaround.

**Comment by pstorch** [comment]
Close + Full unlock didn't work either. I also got the expired token error when closing the DB (maybe it tries to sync again).
I solved it now by choosing "Change database" and open it again from OneDrive.

**Comment by PhilippC** [comment]
sholud be fixed in 1.07-r0 which is now on Google Play.
